### PR TITLE
Fixed warnings related to int-to-string conversions

### PR DESCRIPTION
**Pull request by mitza-oci**

### CODE REVIEW

File: dds/DCPS/SafetyProfileStreams.cpp [hunk 1]
```diff
@@ -14,85 +14,67 @@ OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
...
-  } else {
-    fmt = "%u";
-    const int buff_size = 20 + 1; // note +1 for null terminator
-    char buf[buff_size];
-    ACE_OS::snprintf(&buf[0], buff_size, fmt, to_convert);
-    return String(buf);
-  }
+  static const int buff_size = 20 + 1; // note +1 for null terminator
+  char buf[buff_size];
+  ACE_OS::snprintf(&buf[0], buff_size, as_hex ? "%x" : "%u", to_convert);
```
Comment (jrw972) [inline dds/DCPS/SafetyProfileStreams.cpp:45]: Did you intend to now use `%08x`?

File: tools/rtpsrelay/GuidAddrSet.cpp [hunk 2]
```diff
@@ -281,7 +281,7 @@ void GuidAddrSet::remove(const OpenDDS::DCPS::GUID_t& guid,
   relay_stats_reporter_.local_active_participants(guid_addr_set_map_.size(), now);
 
   if (config_.log_activity()) {
-    ACE_DEBUG((LM_INFO, "(%P|%t) INFO: GuidAddrSet::remove_i "
+    ACE_DEBUG((LM_INFO, "(%P|%t) INFO: GuidAddrSet::remove "
```
Comment (jrw972) [inline tools/rtpsrelay/GuidAddrSet.cpp:284]:
```suggestion
    ACE_DEBUG((LM_INFO, "(%P|%t) INFO: GuidAddrSet::remove: "
```

**Review by jrw972 — CHANGES_REQUESTED** [review]

File: dds/DCPS/SafetyProfileStreams.cpp (hunk 1 above)
Comment (mitza-oci) [reply dds/DCPS/SafetyProfileStreams.cpp:45]: No, I meant to come back to it after I had a better idea of what the formats should be.

File: tools/rtpsrelay/GuidAddrSet.cpp (hunk 2 above)
Comment (mitza-oci) [reply tools/rtpsrelay/GuidAddrSet.cpp:284]: No other logging statement in this file uses a `:` here, so if we're going to change it we should make it consistent either by file or by project.

Comment (jrw972) [reply tools/rtpsrelay/GuidAddrSet.cpp:284]: Oh.  :(. Nevermind then.

**Review by iguessthislldo — APPROVED** [review]

**Comment by mitza-oci** [comment]
- [ ] Update from @jwillemsen comment on #4685 
- [ ] Make hex formats fixed-width using leading 0s
- [ ] Add unit tests

**Review by jrw972 — APPROVED** [review]

**Pull request merged**